### PR TITLE
Alpha Add command for Kyma CLI

### PR DESCRIPTION
**Pull request by Cortey**
**Description**
Adds the "Add" command to CLI allowing to install modules based on list of wanted modules or a custom manifest from path.

Misses the "specify version" functionality and "remove command" which will be added in the following PRs

**Related issue(s)**
https://github.com/kyma-project/cli/issues/2083

### CODE REVIEW

File: internal/cmd/alpha/add/add.go [hunk 1]
```diff
@@ -1,20 +1,196 @@
...
+	return nil
+}
+
+func applySpecifiedModules(cfg *addConfig) clierror.Error {
+	modules, err := getAvailableModules()
+	if err != nil {
+		return err
+	}
+	for _, rec := range modules {
+		if containsModule(rec.Name, cfg.wantedModules) {
```
Comment (anoipm) [inline internal/cmd/alpha/add/add.go:85]:
```suggestion
		if !containsModule(rec.Name, cfg.wantedModules) {
		    continue
		}
```

File: internal/cmd/alpha/add/add.go [hunk 2]
```diff
@@ -1,20 +1,196 @@
...
+			objects, err := decodeYaml(bytes.NewReader(yamlContent))
+			if err != nil {
+				return clierror.Wrap(err, clierror.New("failed to decode YAML"))
+			}
+
+			err = cfg.KubeClient.RootlessDynamic().ApplyMany(cfg.Ctx, objects)
+			if err != nil {
+				return clierror.Wrap(err, clierror.New("failed to apply module resources"))
+			}
+		}
```
Comment (anoipm) [inline internal/cmd/alpha/add/add.go:110]:
```suggestion
```

File: internal/cmd/alpha/add/add.go [hunk 3]
```diff
@@ -1,20 +1,196 @@
...
+	err := assureNamespace("kyma-system", cfg)
+	if err != nil {
+		return err
+	}
+
+	if cfg.custom != "" {
+		err = applyCustomConfiguration(cfg)
+		if err != nil {
+			return err
+		}
```
Comment (anoipm) [inline internal/cmd/alpha/add/add.go:68]:
```suggestion
		return applyCustomConfiguration(cfg)
```

File: internal/cmd/alpha/add/add.go [hunk 4]
```diff
@@ -1,20 +1,196 @@
...
+	} else {
+		err = applySpecifiedModules(cfg)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
```
Comment (anoipm) [inline internal/cmd/alpha/add/add.go:76]:
```suggestion
	}
        return applySpecifiedModules(cfg)
```

File: internal/cmd/alpha/add/add.go [hunk 5]
```diff
@@ -1,20 +1,196 @@
...
+	}
+	defer resp.Body.Close()
+
+	var modules communitymodules.Modules
+	return communitymodules.DecodeCommunityModulesResponse(resp, modules)
+}
+
+func assureNamespace(namespace string, cfg *addConfig) clierror.Error {
+	_, err := cfg.KubeClientConfig.KubeClient.Static().CoreV1().Namespaces().Get(cfg.Ctx, namespace, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
```
Comment (anoipm) [inline internal/cmd/alpha/add/add.go:149]:
```suggestion
	if !errors.IsNotFound(err) {
	    return nil
	}
```

File: internal/cmd/alpha/add/add.go [hunk 6]
```diff
@@ -1,20 +1,196 @@
...
+		_, err = cfg.KubeClientConfig.KubeClient.Static().CoreV1().Namespaces().Create(cfg.Ctx, &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return clierror.New("failed to create namespace")
+		}
+	}
```
Comment (anoipm) [inline internal/cmd/alpha/add/add.go:158]:
```suggestion
```

File: internal/kube/rootlessdynamic/client.go [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+		return split[0], split[1]
+	}
+	return "", split[0]
+}
+
+func kindToResource(kind string) string {
+	return makePlural(strings.ToLower(kind))
+}
+
+func makePlural(singular string) string {
```
[the remark this answers is not in the document]
Comment (Cortey) [reply internal/kube/rootlessdynamic/client.go:71]: They only need to be compatible with our resources, i don't think we need to cover exceptions that usually have nothing to do with nouns used to name resources

File: internal/kube/rootlessdynamic/client.go [hunk 8]
```diff
@@ -55,6 +65,32 @@ func (c *client) ApplyMany(ctx context.Context, objs []unstructured.Unstructured
 	return nil
 }
 
+func (c *client) discoverGVR(apiVersion, kind string) (*schema.GroupVersionResource, error) {
+	group, version := groupVersion(apiVersion)
+	groupVersion := schema.GroupVersion{Group: group, Version: version}
+
+	text, err := c.discovery.ServerResourcesForGroupVersion(groupVersion.String())
```
Comment (pPrecel) [inline internal/kube/rootlessdynamic/client.go:72]: I would suggest to rename this `text` var

File: internal/kube/rootlessdynamic/client.go [hunk 9]
```diff
@@ -55,6 +65,32 @@ func (c *client) ApplyMany(ctx context.Context, objs []unstructured.Unstructured
...
+				Version:  version,
+				Resource: resourceList.Name,
+			}
+			break
+		}
+	}
+	if gvr == nil {
+		return nil, errors.New("Resource " + kind + " in apiVersion " + apiVersion + " not registered on cluster")
+	}
+	return gvr, nil
```
Comment (pPrecel) [inline internal/kube/rootlessdynamic/client.go:91]:
```suggestion
	for _, resourceList := range text.APIResources {
		if resourceList.Kind == kind {
			return &schema.GroupVersionResource{
				Group:    group,
				Version:  version,
				Resource: resourceList.Name,
			}, nil
		}
	}
	
	return nil, errors.New("Resource " + kind + " in apiVersion " + apiVersion + " not registered on cluster")
```

**Pull request merged**